sdk/servicebus/service-bus/test/batchReceiver.spec.ts

```diff
@@ -1135,3 +1135,59 @@ describe("Batch Receiver - Others", function(): void {
     await testAskForMore(true);
   });
 });
+
+describe("Batching - disconnects", function(): void {
+  afterEach(async () => {
+    return afterEachTest();
+  });
+
+  it("can receive and settle messages after a disconnect", async function(): Promise<void> {
+    // Create the sender and receiver.
+    sbClient = getServiceBusClient();
+    const { receiverClient, senderClient } = await getSenderReceiverClients(
+      sbClient,
+      TestClientType.UnpartitionedQueue,
+      TestClientType.UnpartitionedQueue
+    );
+    const receiver = receiverClient.createReceiver(ReceiveMode.peekLock);
+    const sender = senderClient.createSender();
+    // Send a message so we can be sure when the receiver is open and active.
+    await sender.send(TestMessage.getSample());
+
+    let settledMessageCount = 0;
+
+    const messages1 = await receiver.receiveMessages(1, 5);
+    for (const message of messages1) {
+      await message.complete();
+      settledMessageCount++;
+    }
+
+    settledMessageCount.should.equal(1, "Unexpected number of settled messages.");
+
+    const connectionContext = receiver["_context"].namespace;
+    const refreshConnection = connectionContext.refreshConnection;
+    let refreshConnectionCalled = 0;
+    connectionContext.refreshConnection = function(...args) {
+      refreshConnectionCalled++;
+      refreshConnection.apply(this, args);
```

Comment:
That's...beautiful. New technique acquired....

Reply:
It's the same concept as spies (sinon, jasmine, jest have this), except I really didn't want to pull in a new dependency when this is all I needed :)